### DISCUSSION TITLE
Python UPLOAD.EXE being flagged by Malwarebytes as a Trojan

### DISCUSSION THREAD

**Discussion by greenperro** [philosowaffle/peloton-to-garmin#157] [Q&A] [answered]
Good morning.  For the 2nd time now, Malwarebytes has flagged UPLOAD.EXE as a trojan.  Is this a false positive?  If I quarantine the file, will it prevent uploads to Garmin?

**Comment by philosowaffle** [comment, marked as answer]
To the best of my knowledge this is a false positive, however I encourage everyone to not take my word and verify the code for themselves.

Yes, quarantining this file will prevent uploads to Garmin.

[Here ](https://github.com/philosowaffle/peloton-to-garmin/tree/master/python) is the python source code used to preform the upload.  And [here ](https://github.com/philosowaffle/peloton-to-garmin/blob/master/.github/workflows/tag-release.yml#L127) is the tool and command used to transform that python script into a Windows exe.

This exe is provided for the benefit of the less tech savvy who just want an exe they can run without having to mess with additionally installing dependencies on their system like python or dotnet.

If anyone still has concerns about this file and would like to avoid using it, you can modify [the UploadStrategy value in your config file](https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration#garmin-config).  Switching to either `0` or `1` will remove all dependencies on this `upload.exe` file allowing you to safely remove it.

**Reply by greenperro** [reply]
Thanks very much...  I have a feeling it is not really anything.  I also appreciate the info about how to disable it if needed.
